services/compute/baremetalmachine/baremetalmachine.go

```diff
@@ -0,0 +1,379 @@
+			return nil, errors.Wrapf(errors.InvalidInput, "SSH KeyData is missing")
+		}
+		keys = append(keys, &wssdcloudcompute.SSHPublicKey{Keydata: *key.KeyData})
+	}
+	return keys, nil
+
+}
+
+func (c *client) getWssdBareMetalMachineLinuxConfiguration(linuxConfiguration *compute.LinuxConfiguration) *wssdcloudcompute.LinuxConfiguration {
+	lc := &wssdcloudcompute.LinuxConfiguration{}
+
+	if linuxConfiguration.DisablePasswordAuthentication != nil {
+		lc.DisablePasswordAuthentication = *linuxConfiguration.DisablePasswordAuthentication
+	}
+
+	return lc
+
+}
+
+func (c *client) getWssdBareMetalMachineOSConfiguration(s *compute.BareMetalMachineOSProfile) (*wssdcloudcompute.BareMetalMachineOperatingSystemConfiguration, error) {
+	publickeys := []*wssdcloudcompute.SSHPublicKey{}
+	var err error
+
+	osconfig := wssdcloudcompute.BareMetalMachineOperatingSystemConfiguration{
+		Users: []*wssdcloudcompute.UserConfiguration{},
+	}
+
+	if s == nil {
+		return &osconfig, nil
+	}
+
+	if s.LinuxConfiguration != nil {
+		var sshConfiguration *compute.SSHConfiguration = s.LinuxConfiguration.SSH
+
+		if sshConfiguration != nil {
+			publickeys, err = c.getWssdBareMetalMachineOSSSHPublicKeys(sshConfiguration)
+			if err != nil {
+				return nil, errors.Wrapf(err, "SSH Configuration Invalid")
+			}
+		}
+	}
+
+	adminuser := &wssdcloudcompute.UserConfiguration{}
+	if s.AdminUsername != nil {
+		adminuser.Username = *s.AdminUsername
+	}
+
+	if s.AdminPassword != nil {
+		adminuser.Password = *s.AdminPassword
+	}
+
+	if s.ComputerName == nil {
+		return nil, errors.Wrapf(errors.InvalidInput, "ComputerName is missing")
+	}
+
+	osconfig.ComputerName = *s.ComputerName
+	osconfig.Administrator = adminuser
+	osconfig.Publickeys = publickeys
+
+	if s.LinuxConfiguration != nil {
+		osconfig.LinuxConfiguration = c.getWssdBareMetalMachineLinuxConfiguration(s.LinuxConfiguration)
+	}
+
+	if s.CustomData != nil {
+		osconfig.CustomData = *s.CustomData
+	}
+	return &osconfig, nil
+}
+
+// Conversion functions from wssdcloudcompute to compute
+
+func (c *client) getBareMetalMachine(bmm *wssdcloudcompute.BareMetalMachine, group string) *compute.BareMetalMachine {
+	return &compute.BareMetalMachine{
+		Name: &bmm.Name,
+		ID:   &bmm.Id,
+		Tags: getComputeTags(bmm.GetTags()),
+		BareMetalMachineProperties: &compute.BareMetalMachineProperties{
+			ProvisioningState: status.GetProvisioningState(bmm.GetStatus().GetProvisioningStatus()),
+			Statuses:          c.getBareMetalMachineStatuses(bmm),
+			StorageProfile:    c.getBareMetalMachineStorageProfile(bmm.Storage),
+			HardwareProfile:   c.getBareMetalMachineHardwareProfile(bmm),
+			SecurityProfile:   c.getBareMetalMachineSecurityProfile(bmm),
+			OsProfile:         c.getBareMetalMachineOSProfile(bmm.Os),
+			NetworkProfile:    c.getBareMetalMachineNetworkProfile(bmm.Network),
+			Host:              c.getBareMetalMachineHostDescription(bmm),
+		},
+		Version:  &bmm.Status.Version.Number,
+		Location: &bmm.LocationName,
```

Comment:
This seems to be missing GroupName and NodeName. Is that on purpose?

Reply:
NodeName is contained in Host. This is how it is for the VM and I followed. GroupName is not defined in the SDK. Note that group is a param to the API. I believe for a VM, both location and group have to be present. I assummed the same for a bare metal machine.

Reply:
Sounds good, lets keep it symmetric for now.